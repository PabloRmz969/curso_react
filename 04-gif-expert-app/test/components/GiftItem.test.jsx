import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en gifitem', () => {
    const title = 'Naruto';
    const url = 'https://naruto.jpg/';

    test('debe hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    })

    test('debe mostrar el alt y src de la imagen', () => {
        render(<GifItem title={title} url={url} />);
        //screen.debug();
        //expect(screen.getByRole('img').src).toBe(url);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('debe mostar el tÃ­tulo del componente', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy()

    })
})