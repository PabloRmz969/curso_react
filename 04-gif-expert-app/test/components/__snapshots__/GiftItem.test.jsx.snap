// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en gifitem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naruto"
      src="https://naruto.jpg/"
    />
    <p>
      Naruto
    </p>
  </div>
</div>
`;
